### ISSUE TITLE
Issue regarding uda loss function

### ISSUE THREAD

**Issue by getsanjeev** [peisuke/UnsupervisedDataAugmentation.pytorch#2]
Why do we calculate KL divergence between softmax and log of softmax distributions?
pred_x = F.softmax(model(x), dim=1)
pred_x_h = F.log_softmax(model(x_h), dim=1)
Why not just softmax of both predictions.
Any response would be helpful.

**Comment by nazim1021** [comment]
Hi sanjeev, 
Its because the KL divergence implemented in pytorch expects the one of the input to be log of probabilites
https://pytorch.org/docs/stable/nn.html#torch.nn.KLDivLoss

**Comment by getsanjeev** [comment]
Thank you, that seems right.
